src/components/SignUp/index.tsx

```diff
@@ -129,10 +128,8 @@ const SignUpFormComponent: React.FC<SignUpFormProps> = ({
     const { formatMessage } = useIntl();
 
     const disableButton = React.useCallback((): boolean => {
-        if (!hasConfirmed || isLoading || !email.match(EMAIL_REGEX) || !password || !confirmPassword) {
-            if (isNicknamesEnabled() && !nickname.match(NICKNAME_REGEX)) {
-                return false;
-            }
+        if (!hasConfirmed || isLoading || !email.match(EMAIL_REGEX) || !password || !confirmPassword ||
+        (isNicknamesEnabled() && !nickname.match(NICKNAME_REGEX))) {
```

Comment:
```suggestion
        if (!hasConfirmed || isLoading || !email.match(EMAIL_REGEX) || !password || !confirmPassword ||
           (isNicknamesEnabled() && !nickname.match(NICKNAME_REGEX))) {
```
fix indentation here